### ISSUE TITLE
Trouble adding detectors to table when using FCCD statistics

### ISSUE THREAD

**Issue by ambarb** [NSLS-II/bluesky#400]
More details to come.

```
In [134]: gs.DETS
Out[134]: 
[EpicsMotor(prefix='XF:23ID1-ES{Dif-Ax:Th}Mtr', name='theta', settle_time=0.0, timeout=None, read_attrs=['user_readback', 'user_setpoint'], configuration_attrs=['motor_egu'], monitor_attrs=[]),
 EpicsMotor(prefix='XF:23ID1-ES{Dif-Ax:Del}Mtr', name='delta', settle_time=0.0, timeout=None, read_attrs=['user_readback', 'user_setpoint'], configuration_attrs=['motor_egu'], monitor_attrs=[]),
 EpicsMotor(prefix='XF:23ID1-ES{Dif-Ax:Gam}Mtr', name='gamma', settle_time=0.0, timeout=None, read_attrs=['user_readback', 'user_setpoint'], configuration_attrs=['motor_egu'], monitor_attrs=[]),
 EpicsMotor(prefix='XF:23ID1-ES{Dif-Ax:X}Mtr', name='sx', settle_time=0.0, timeout=None, read_attrs=['user_readback', 'user_setpoint'], configuration_attrs=['motor_egu'], monitor_attrs=[]),
 EpicsMotor(prefix='XF:23ID1-ES{Dif-Ax:Y}Mtr', name='say', settle_time=0.0, timeout=None, read_attrs=['user_readback', 'user_setpoint'], configuration_attrs=['motor_egu'], monitor_attrs=[]),
 EpicsMotor(prefix='XF:23ID1-ES{Dif-Ax:Z}Mtr', name='saz', settle_time=0.0, timeout=None, read_attrs=['user_readback', 'user_setpoint'], configuration_attrs=['motor_egu'], monitor_attrs=[]),
 Cryoangle(prefix='', name='cryoangle', settle_time=0.0, timeout=None, read_attrs=['readback', 'setpoint'], configuration_attrs=[], monitor_attrs=[], limits=None, egu=''),
 SamplePosVirtualMotor(prefix='XF:23ID1-ES{Dif-Ax:SY}', name='sy', settle_time=0.0, timeout=None, read_attrs=['readback', 'setpoint'], configuration_attrs=[], monitor_attrs=[], limits=None, egu=''),
 SamplePosVirtualMotor(prefix='XF:23ID1-ES{Dif-Ax:SZ}', name='sz', settle_time=0.0, timeout=None, read_attrs=['readback', 'setpoint'], configuration_attrs=[], monitor_attrs=[], limits=None, egu=''),
 Temperature(prefix='XF:23ID1-ES{TCtrl:1', name='temp', read_attrs=['a', 'b'], configuration_attrs=[], monitor_attrs=[]),
 EpicsSignal(read_pv='UT:SB1-Cu:1{}T:Spply_Ld-I', name='uw_temp', value=85.15884399414062, timestamp=1459451393.579187, pv_kw={}, auto_monitor=False, string=False, write_pv='UT:SB1-Cu:1{}T:Spply_Ld-I', limits=False, put_complete=False),
 PGMEnergy(prefix='XF:23ID1-OP{Mono', name='pgm_en', parent='pgm', settle_time=0.0, timeout=None, read_attrs=['readback', 'setpoint', 'stop_signal'], configuration_attrs=[], monitor_attrs=[], limits=None, egu=''),
 EPU1(prefix='XF:23ID-ID{EPU:1', name='epu1', read_attrs=['gap', 'phase'], configuration_attrs=[], monitor_attrs=[]),
 EPU2(prefix='XF:23ID-ID{EPU:2', name='epu2', read_attrs=['gap', 'phase'], configuration_attrs=[], monitor_attrs=[]),
 SlitsGapCenter(prefix='XF:23ID1-OP{Slt:1', name='slt1', read_attrs=['xg', 'xc', 'yg', 'yc'], configuration_attrs=[], monitor_attrs=[]),
 SlitsGapCenter(prefix='XF:23ID1-OP{Slt:2', name='slt2', read_attrs=['xg', 'xc', 'yg', 'yc'], configuration_attrs=[], monitor_attrs=[]),
 SlitsXY(prefix='XF:23ID1-OP{Slt:3', name='slt3', read_attrs=['x', 'y'], configuration_attrs=[], monitor_attrs=[]),
 Mirror(prefix='XF:23IDA-OP:1{Mir:1', name='m1a', read_attrs=['z', 'y', 'x', 'pit', 'yaw', 'rol'], configuration_attrs=[], monitor_attrs=[]),
 MotorMirror(prefix='XF:23ID1-OP{Mir:3', name='m3a', read_attrs=['x', 'pit', 'bdr'], configuration_attrs=[], monitor_attrs=[]),
 EpicsSignal(read_pv='XF:23ID1-OP{TCtrl:1-Chan:A}T-I', name='mono_tempa', value=302.79, timestamp=1459451392.400664, pv_kw={}, auto_monitor=False, string=False, write_pv='XF:23ID1-OP{TCtrl:1-Chan:A}T-I', limits=False, put_complete=False),
 EpicsSignal(read_pv='XF:23ID1-OP{TCtrl:1-Chan:B}T-I', name='mono_tempb', value=302.76, timestamp=1459451392.563194, pv_kw={}, auto_monitor=False, string=False, write_pv='XF:23ID1-OP{TCtrl:1-Chan:B}T-I', limits=False, put_complete=False),
 EpicsSignal(read_pv='XF:23ID1-OP{Mon-Grt:1}T-I', name='grt1_temp', value=31.900000000000002, timestamp=1459451393.050221, pv_kw={}, auto_monitor=False, string=False, write_pv='XF:23ID1-OP{Mon-Grt:1}T-I', limits=False, put_complete=False),
 EpicsSignal(read_pv='XF:23ID1-OP{Mon-Grt:2}T-I', name='grt2_temp', value=28.700000000000003, timestamp=1459451393.050266, pv_kw={}, auto_monitor=False, string=False, write_pv='XF:23ID1-OP{Mon-Grt:2}T-I', limits=False, put_complete=False),
 EpicsSignal(read_pv='XF:23ID-SR{}I-I', name='ring_curr', value=228.76019071999997, timestamp=1459451392.869853, pv_kw={}, auto_monitor=False, string=False, write_pv='XF:23ID-SR{}I-I', limits=False, put_complete=False),
 EpicsScaler(prefix='XF:23ID1-ES{Sclr:1}', name='sclr', read_attrs=['channels', 'time'], configuration_attrs=['preset_time', 'presets', 'gates'], monitor_attrs=[]),
 ProductionCam(prefix='XF:23ID1-ES{FCCD}', name='fccd', read_attrs=['hdf5', 'acquire_time', 'num_images_captured', 'plugin_num_images', 'stats1', 'stats2', 'stats3', 'stats4', 'stats5'], configuration_attrs=[], monitor_attrs=[])]

In [135]: gs.TABLE_COLS = ['sclr_ch2','epu1_gap_readback','mono_tempa','mono_tempb','grt1_temp','grt2_temp','uw_temp'];

In [136]: gs.PLOT_Y = 'sclr_ch2'

In [137]: ct()
Transient Scan ID: 60921
Persistent Unique Scan ID: 'b096b9dc-d0df-493b-8a0a-70b20f8e78e5'
+-----------+------------+------------+-------------------+------------+------------+------------+------------+------------+
|   seq_num |       time |   sclr_ch2 | epu1_gap_readback | mono_tempa | mono_tempb |  grt1_temp |  grt2_temp |    uw_temp |
+-----------+------------+------------+-------------------+------------+------------+------------+------------+------------+
|         1 | 15:10:29.3 |         44 |           27.4141 |    302.780 |    302.750 |       31.8 |       28.7 |      85.13 |
+-----------+------------+------------+-------------------+------------+------------+------------+------------+------------+
Count ['b096b9'] (scan num: 60921)
Out[137]: ['b096b9dc-d0df-493b-8a0a-70b20f8e78e5']

In [138]: 
KeyboardInterrupt

In [138]: 

In [138]: 

In [138]: gs.TABLE_COLS = ['sclr_ch2','epu1_gap_readback','diag6_monitor','mono_tempa','mono_tempb','grt1_temp','grt2_temp','uw_temp'];

In [139]: gs.PLOT_Y = 'sclr_ch2'

In [140]: ct()
Transient Scan ID: 60922
Persistent Unique Scan ID: '07dd3228-9705-44cf-8154-460d2af35610'
+-----------+------------+------------+-------------------+------------+------------+------------+------------+------------+
|   seq_num |       time |   sclr_ch2 | epu1_gap_readback | mono_tempa | mono_tempb |  grt1_temp |  grt2_temp |    uw_temp |
+-----------+------------+------------+-------------------+------------+------------+------------+------------+------------+
|         1 | 15:11:11.3 |         47 |           27.4141 |    302.750 |    302.740 |       32.1 |       28.8 |      85.07 |
+-----------+------------+------------+-------------------+------------+------------+------------+------------+------------+
Count ['07dd32'] (scan num: 60922)
Out[140]: ['07dd3228-9705-44cf-8154-460d2af35610']

In [141]: diag6_monitor = EpicsSignal('XF:23ID1-BI{Diag:6-Cam:1}Stats1:Total_RBV',
   .....:                             name='diag6_monitor')

In [142]:                             

In [142]: 

In [142]: 

In [142]: 

In [142]: 

In [142]: 

In [142]: gs.TABLE_COLS = ['sclr_ch2','epu1_gap_readback','diag6_monitor','mono_tempa','mono_tempb','grt1_temp','grt2_temp','uw_temp'];

In [143]: gs.PLOT_Y = 'sclr_ch2'

In [144]: ct()
Transient Scan ID: 60923
Persistent Unique Scan ID: 'c1677d02-6ff8-42cf-beaa-346d1f5a9045'
+-----------+------------+------------+-------------------+------------+------------+------------+------------+------------+
|   seq_num |       time |   sclr_ch2 | epu1_gap_readback | mono_tempa | mono_tempb |  grt1_temp |  grt2_temp |    uw_temp |
+-----------+------------+------------+-------------------+------------+------------+------------+------------+------------+
|         1 | 15:12:09.4 |         45 |           27.4141 |    302.710 |    302.720 |       32.2 |       28.8 |      85.01 |
+-----------+------------+------------+-------------------+------------+------------+------------+------------+------------+
Count ['c1677d'] (scan num: 60923)
Out[144]: ['c1677d02-6ff8-42cf-beaa-346d1f5a9045']

In [145]: gs.TABLE_COLS = ['sclr_ch2','epu1_gap_readback','diag6_monitor','mono_tempa','mono_tempb','grt1_temp','grt2_temp'];

In [146]: ct()
Transient Scan ID: 60924
Persistent Unique Scan ID: '8f2821a6-2d09-4002-971e-0517cabdcf89'
+-----------+------------+------------+-------------------+------------+------------+------------+------------+
|   seq_num |       time |   sclr_ch2 | epu1_gap_readback | mono_tempa | mono_tempb |  grt1_temp |  grt2_temp |
+-----------+------------+------------+-------------------+------------+------------+------------+------------+
|         1 | 15:12:31.3 |         49 |           27.4141 |    302.720 |    302.740 |       32.1 |       28.7 |
+-----------+------------+------------+-------------------+------------+------------+------------+------------+
Count ['8f2821'] (scan num: 60924)
Out[146]: ['8f2821a6-2d09-4002-971e-0517cabdcf89']

In [147]: gs.TABLE_COLS = ['fccd_stats1_total','fccd_stats2_total','fccd_stats3_total',' epu1_gap_readback','mono_tempa','mono_tempb','grt1_temp','grt2_temp'];

In [148]: gs.PLOT_Y = 'fccd_stats1_total'

In [149]: fccd_set(0.5, 6)
Current fCCD config: 3.0 images @ 0.50 Hz = 6.0 s

In [150]: ct()
Transient Scan ID: 60925
Persistent Unique Scan ID: '6be2ef2f-6e02-428a-b44e-7f9cd6a83fb2'
+-----------+------------+-------------------+-------------------+-------------------+------------+------------+------------+------------+
|   seq_num |       time | fccd_stats1_total | fccd_stats2_total | fccd_stats3_total | mono_tempa | mono_tempb |  grt1_temp |  grt2_temp |
+-----------+------------+-------------------+-------------------+-------------------+------------+------------+------------+------------+
|         1 | 15:13:33.3 |             15681 |             33396 |           -310917 |    302.720 |    302.730 |       32.2 |       28.8 |
+-----------+------------+-------------------+-------------------+-------------------+------------+------------+------------+------------+
Count ['6be2ef'] (scan num: 60925)
Out[150]: ['6be2ef2f-6e02-428a-b44e-7f9cd6a83fb2']

In [151]: 
```

**Comment by tacaswell** [comment]
What is the problem?

**Comment by ericdill** [comment]
Looks like `epu1_gap_readback` is added to the table cols in `[147]` but is not shown in the table in `[150]`

`'diag6_monitor'` is set in `[142]` and `[145]` but not shown in `[144]` and `[146]`

Maybe that is what @ambarb is talking about?

**Comment by danielballan** [comment]
I believe that 'epu_gap_readback' may have been renamed to just plain 'epu_gap' in their configuration. @ambarb, if this issue is still a problem, please re-open and help us understand the issue better.

**Comment by ambarb** [comment]
The issue here was that when I added the ````fccd_stats```` to table columns that I could not add the epu  (````epu1_gap_readback```` is correct for the configuration files we used at the time because it works with ````sclr_ch2````).

I tested it just  now. It is fixed. Sorry for the unclear bug report. :(

**Comment by danielballan** [comment]
No problem! Thanks for the follow-up.

Get Outlook for iOS<https://aka.ms/o0ukef>




On Sat, Jun 4, 2016 at 7:27 PM -0700, "ambarb" <notifications@github.com<mailto:notifications@github.com>> wrote:


The issue here was that when I added the fccd_stats to table columns that I could not add the epu (epu1_gap_readback is correct for the configuration files we used at the time because it works with sclr_ch2).

I tested it just now. It is fixed. Sorry for the unclear bug report. :(
